### ISSUE TITLE
Sim-to-Real Mismatch: UR Robot Discrepancy in Simulation vs Real World

### ISSUE THREAD

**Issue by Cheng909865905** [vlc-robot/robot-3dlotus#16]
Great work! We noticed that the robot arm used in the simulation differs from the one used in real-world deployment. Could you clarify the reason for this discrepancy? Does it impact manipulation performance? Additionally, could you provide a detailed guide on how to deploy the UR robot for real-world experiments?

**Comment by rjgpinel** [comment]
Hi @Cheng909865905,

Thank you for your interest on 3D-LOTUS. The discrepancy of UR5 robot for the real robot and Franka Emika Panda in simulation is just a matter of the available resources.

- Could you clarify the reason for this discrepancy? 
RLBench was originally designed with the Franka Emika Panda robot, and then, we decided to maintain the original robot and just create GEMBench around the RLBench framework. Then, to showcase the results on the real robot, we use UR5 equipped with the RG-6 gripper, because it was available in our lab (we currently don't have a Franka Emika Panda robot in our lab).

-  Does it impact manipulation performance?
We finetune our policies on real robot data to make it work on the robot. I believe that using the Panda robot as the real robot, might be helpful and make easier to try to perform sim-to-real transfer, but still the real world might have other discrepancies like the depth sensors noise, lighting conditions, table texture, objects realism, etc. So we might still need to finetune on real robot data to make it work.

Could you provide a detailed guide on how to deploy the UR robot for real-world experiments?
I am sorry but a detailed guide on how to deploy the UR robot for real-world experiments might be very difficult. It depends a lot in your specific robotic setup. In our case, we are using moveit and ROS to control the robot. And you could take a look at our public ROS stack here: https://github.com/inria-paris-robotics-lab/prl_ur5_robot 

I think this might be useful for you to deploy your UR robot.

**Comment by Cheng909865905** [comment]
Thank you for your detailed explanation. I have one additional question:

In your paper, you mentioned that in simulation, you use K = 4 cameras positioned at the front, left shoulder, right shoulder, and wrist.
However, in real-world experiments, you state that "our real robot setup includes a 6-DoF UR5 robotic arm equipped with three RealSense D435 cameras."

Why is there a discrepancy in the number of cameras used between simulation and the real-world setup?

Also, I currently only have access to two depth cameras. In your opinion, what would be the most effective placement for them?
For example:

One in front of the robot and one on the wrist?

One overhead and one on the wrist?

Or another configuration?

I'd appreciate your advice!

**Comment by Cheng909865905** [comment]
@rjgpinel

**Comment by rjgpinel** [comment]
- Why is there a discrepancy in the number of cameras used between simulation and the real-world setup?
Here the reason is the same as why there is a discrepancy between the used robot: available resources and how overcomplicated it is to use 4 cameras vs the improvement in the final performance. In any case, we just want to compare methods in the same setup, and finetuning is required since the setup is different (different objects, lighting, sensors noise, robot, and many others).

- I currently only have access to two depth cameras. In your opinion, what would be the most effective placement for them?
A wrist camera is always helpful to have a more precise manipulation when you approach an object because you can have close-up views or cope better with occlusions by moving the arm. However, it has some drawbacks: (1) there will be a cable that supply energy and connects the camera with your computer, so it might make motion planning more difficult and the cable could collide with objects if it's not properly set, (2) a wrist camera might suffer from vibrations (3) if it's not well placed you might not see the gripper or the viewpoint might not be useful (viewpoint too close to the gripper or just showing the empty table most of the time). So it can be very useful but you need to set this camera well at a correct location with respect to your gripper/arm.

Another option would be to have two external cameras with a rotation of 90º between both, and some inclination to have objects height information (this way you could try to have viewpoints that cover the three x, y, z axis).

This configuration is really up to you and it depends a lot in your use case, resources, setting, etc.

- What exactly is the tele-operation device mentioned in the paper? (Is it a remote controller, gamepad, or VR device?
We are using the Logitech F710 and the joy ROS library  (https://wiki.ros.org/joy) to map buttons to end-effector velocities, gripper openness, etc.
